src/component/TodoForm.tsx

```
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  todos: Todo[];
+  setErrorMessage: (message: string) => void;
+  onAddTodo: (value: string) => Promise<void>;
+  inputRef: React.RefObject<HTMLInputElement> | null;
+  todosLength: number;
+  isTitleDisabled: boolean;
+  toggleAll: () => Promise<void>;
+};
+
+export const TodoForm: React.FC<Props> = ({
+  todos,
+  setErrorMessage,
+  onAddTodo,
+  inputRef,
+  todosLength,
+  isTitleDisabled,
+  toggleAll,
+}) => {
+  const [title, setTitle] = useState<string>('');
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (title.trim() === '') {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    try {
+      await onAddTodo(title.trim());
+      setTitle('');
+    } catch (error) {}
```

It's good that you have a try-catch block, but you're not handling the error in the `catch` block. At a minimum, you should probably log the error or display a user-friendly message.  Consider whether simply swallowing the error is the desired behavior.